asapdiscovery-data/asapdiscovery/data/schema/ligand.py

```diff
@@ -379,16 +432,44 @@ def to_sdf(self, filename: Union[str, Path], allow_append=False) -> None:
         mol = self.to_oemol()
         write_file_directly(filename, oemol_to_sdf_string(mol), mode=fmode)
 
-    def set_SD_data(self, data: dict[str, str]) -> None:
+    def set_SD_data(self, data: dict[str, Union[str, list]]) -> None:
         """
         Set the SD data for the ligand, uses an update to overwrite existing data in line with
         OpenEye behaviour
         """
         # make sure we don't overwrite any attributes
-        for k in data.keys():
+        new_data = {}
+        for k, v in data.items():
             if k in self.__fields__.keys():
-                raise ValueError(f"Tag name {k} is a reserved attribute name")
-        self.tags.update(data)
+                warnings.warn(f"Tag name {k} is a reserved attribute name, skipping")
+            else:
+                new_data[k] = v
+
+        # first update the conformer tags, which is simple if the data is a dict of lists
+        if all([isinstance(v, list) for v in new_data.values()]):
```

Comment:
I don't really understand this bit (apologies), can you clarify?

Reply:
this could be simplified by using a single `set_SD_data` openeye function that handles this logic, and then go through an OE object, but here i'm checking to make sure that all of the data is either a dict[str, str] or that it is all dict[str, list]. I'll try implementing this differently so that its a little clearer and assumes that the dict should always be `dict[str, list]` which if the list is len=1, it will update all confs with that value.